echo.go

```diff
@@ -0,0 +1,75 @@
+package lifxlan
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"errors"
+	"math/rand"
+	"net"
+)
+
+type RawEchoResponsePayload struct {
+	Echoing [64]byte
+}
+
+func (d *device) Echo(ctx context.Context, conn net.Conn) error {
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
+
+	if conn == nil {
+		newConn, err := d.Dial()
+		if err != nil {
+			return err
+		}
+		defer newConn.Close()
+		conn = newConn
+
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
+	}
+
+	payload := make([]byte, 64)
+	rand.Read(payload)
```

Comment:
I don't particularly like the idea to always read from `math/read`, as that requires proper seeding (by default it's always seeded with 1).

I would prefer to ask the caller to pass in a `[]byte` into this function, we could auto fill the rest with `math/rand` if the `[]byte` passed in is less than 64 bytes (for example they pass in `nil`, then we just fill all 64 with random bytes). what do you think?

Reply:
That sounds perfect.